src/xo-app/vm/tab-advanced.js

```diff
@@ -113,6 +116,51 @@ class AffinityHost extends Component {
   }
 }
 
+@connectStore(() => ({
+  container: createGetObject((_, { vm }) => vm.$container)
+}))
+class CoresPerSocket extends Component {
+  _updateXenCoresPerSocketValue = createSelector(
+    (maxCoresPerSocket, vCPUs) => getCoresPerSocketPossibilities(maxCoresPerSocket, vCPUs),
```

Comment:
`getCoresPerSocketPossibilities` returns a new array each call, you should memoize it.